{\rtf1\ansi\ansicpg1252\cocoartf1504\cocoasubrtf600
{\fonttbl\f0\fmodern\fcharset0 Courier;}
{\colortbl;\red255\green255\blue255;}
{\*\expandedcolortbl;\csgray\c100000;}
\margl1440\margr1440\vieww16900\viewh8400\viewkind0
\deftab720
\pard\pardeftab720\partightenfactor0

\f0\fs26 \cf0 \expnd0\expndtw0\kerning0
--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700\
2017-02-02  Rishabh Jain  <rishabhj@lnxsrv09.seas.ucla.edu>\
\
	* timespec.h (timespec_cmp): We notice overflow in extreme cases of the\
	differences between a and b, and so instead we just simply do direct\
	comparisons and return values as described by the comments.\
\
+++ coreutils-with-bug-new/lib/timespec.h	2017-02-02 20:50:59.876420000 -0800\
@@ -45,8 +45,9 @@\
 static inline int\
 timespec_cmp (struct timespec a, struct timespec b)\
 \{\
-  int diff = a.tv_sec - b.tv_sec;\
-  return diff ? diff : a.tv_nsec - b.tv_nsec;\
+  if(a.tv_sec < b.tv_sec) \{ return -1;\}\
+  else if(a.tv_sec > b.tv_sec) \{ return 1;\}\
+  else \{ return 0; \}\
 \}\
 \
 # if ! HAVE_DECL_NANOSLEEP}